### DISCUSSION TITLE
How to use with Next js

### DISCUSSION THREAD

**Discussion by cfortuner** [pmndrs/jotai#1410] [Q&A]
Hey!

I'm trying to get a atom value in a child component in my next app. I'm using ssr and the atom isn't available in the child (which is in a ssr page)

Structure is like this:
```
app (sets the atom)
   ... children
    page
       Child (attempts to use the atom)
   ```
   
But the child never sees the atom change.
   
Can I get this to work with jotai + next?

I tried the useHydrationAtom, but that didn't seem to work either  

anyways let me know! Thanks.

**Comment by dai-shi** [comment]
Can you show more code? (but please keep it small)
How do you set atom values in app?

**Reply by cfortuner** [reply]
```
 // _app.ts
   const myBoolAtom = atom(false)
   
   function App() {
        const  setMyBool = useUpdateAtom(myBoolAtom)
        useEffect(() => {
          setMyBool(true)
          }, [])
        
        return (
           //provider and children        
           )
   }
   
   //... child component of a page using ssr 
   
    function Child() {
        const myBool = useAtom(myBoolAtom)
        
        useEffect(() => {
           console.log(myBool) // always false
         }, [myBool])
    }
    ```

**Reply by dai-shi** [reply]
Thanks.
The thing is, your `const  setMyBool = useUpdateAtom(myBoolAtom)` is outside your provider. You want to have one more component nested.
```jsx
const myBoolAtom = atom(false)

const App = () => (
  <Provider>
    <InnerApp />
  </Provider>
)

const InnerApp = () => {
  const setMyBool = useUpdateAtom(myBoolAtom)
  useEffect(() => {
    setMyBool(true)
  }, [])
  return (
    // children
  )
}
```

**Reply by cfortuner** [reply]
oops! that makes sense. thanks!
